avm/res/key-vault/vault/key/main.bicep

```diff
@@ -5,59 +5,8 @@ metadata owner = 'Azure/module-maintainers'
 @description('Conditional. The name of the parent key vault. Required if the template is used in a standalone deployment.')
 param keyVaultName string
 
-@description('Required. The name of the key.')
-param name string
-
-@description('Optional. Resource tags.')
-param tags object?
-
-@description('Optional. Determines whether the object is enabled.')
-param attributesEnabled bool = true
-
-@description('Optional. Expiry date in seconds since 1970-01-01T00:00:00Z. For security reasons, it is recommended to set an expiration date whenever possible.')
-param attributesExp int?
-
-@description('Optional. Not before date in seconds since 1970-01-01T00:00:00Z.')
-param attributesNbf int?
-
-@description('Optional. The elliptic curve name.')
-@allowed([
-  'P-256'
-  'P-256K'
-  'P-384'
-  'P-521'
-])
-param curveName string = 'P-256'
-
-@description('Optional. Array of JsonWebKeyOperation.')
-@allowed([
-  'decrypt'
-  'encrypt'
-  'import'
-  'sign'
-  'unwrapKey'
-  'verify'
-  'wrapKey'
-])
-param keyOps array?
-
-@description('Optional. The key size in bits. For example: 2048, 3072, or 4096 for RSA.')
-param keySize int?
-
-@description('Optional. The type of the key.')
-@allowed([
-  'EC'
-  'EC-HSM'
-  'RSA'
-  'RSA-HSM'
-])
-param kty string = 'EC'
-
-@description('Optional. Array of role assignments to create.')
-param roleAssignments roleAssignmentType
-
-@description('Optional. Key rotation policy properties object.')
-param rotationPolicy object?
+@description('Optional. Sets the attributes of the secret.')
+param keyProperties keyType
```

Comment:
This is a pretty major change to the convention of how modules are written - and I would expect that many of our tests would not be compatible if changed without previous CI updates.

Overall I'm not sure if we'd want a UDT of an entire resource in that resource's module. As you might have noticed yourself, referencing the properties etc. get's quite messy, defaults cannot be set in the type and hence are also not picked up by the calling module as defaults anymore.

I'd suggest to brainstorm this a bit, but from a first impression I'd suggest to only have the UDT in the calling module (and in the future once out of preview store it in a types file) for that type (e.g. `keyType` in the key vault's `/vault/main.bicep` but not the `/vault/key/main.bicep`).

Don't change anything now, but have a chat - ideally including also @ChrisSidebotham & @eriqua as this should be a references for other contributors too.

Reply:
I see what you mean, main goal with my implementation was to do it similarly as we did with `vault/accessPolicies` already, too. I think we should define a line up till which it makes sense to move stuff into a UDT, what do you think?

Reply:
I like the approach, @ChrisSidebotham !
should I do the same for the keyvault module then, @AlexanderSehr ?

Reply:
> I like the approach, @ChrisSidebotham ! should I do the same for the keyvault module then, @AlexanderSehr ?

Its my prefered approach as that way (when) we get functionality to deploy child resources, they are still fully functioning modules and the type is to assist someone who is deploying the parent resources with interfaces or child modules.

Reply:
Agreed. The UDT would 'only' be in the parent.

Mind you, in the future, when we can import resource-udts as well as custom-udts from an external file, we can import nested properties via this central file instead of repeating it potentially over and over. Not relevant yet, but will be eventually.

Reply:
Nothing to add from my side. 100% agree with the suggested approach. Leverage UDT for child modules in the parent, but keep the first line of individual input parameters at child module level.

Reply:
@fblix - Are you okay to see this through?

Reply:
I have ported these changes to #1986